docs/guides/create-histogram.md

```diff
@@ -0,0 +1,192 @@
+
+```bash
+nitric new
+```
+
+Create a project, name it and select your preferred starter template.
+
+```bash
+? What is the name of the project? histogram-api
+? Choose a template: official/Python - Starter
+```
+
+Next, open the project in your editor of choice.
+
+```bash
+> cd histogram-api
+```
+
+Make sure all dependencies are resolved using Pipenv:
+
+```bash
+pipenv install --dev
+```
+
+Starting from scratch in the `hello.py` function file, lets start by importing and defining our api.
+
+```python
+from nitric.resources import api
+from nitric.application import Nitric
+
+mainApi = api("main")
+
+Nitric.run()
+```
+
+We can then define our api route which will accept the histogram data to then be returned as an image.
+
+```python
+from nitric.resources import api
+from nitric.application import Nitric
+
+mainApi = api("main")
+
+@mainApi.get("/histogram")
+async def create_histogram(ctx):
+  pass
+
+Nitric.run()
+```
+
+The route will be registered for a GET method so that it is a link that can be embed. For example, once the api is complete, it will be able to be used in situations like:
+
+```html
+<img
+  src="http://localhost:4001/histogram?data=1,1,2,2,2,3,3,3,3,5,6,4,5,3,2,6,6,6,5,4&ylabel=frequency&xlabel=dice rolls&title=Experimental probablity of 20 dice rolls"
+/>
+```
+
+We can then write the logic for creating the histogram from the request. As seen in the above url, there are 4 query parameters that we want a user to be able to configure, the data, y-label, x-label, and title.
+
+```python
+@mainApi.get("/histogram")
+async def create_histogram(ctx):
+  # Extract the comma-delimited data and split it into array
+  data = ctx.req.query.get("data")[0].split(',')
+
+  # Get the optional histogram labels
+  xlabel = ctx.req.query.get("xlabel")
+  ylabel = ctx.req.query.get("ylabel")
+  title = ctx.req.query.get("title")
+
+  try:
+    data = [int(d) for d in data]
+    data.sort()
+  except ValueError:
+    # If casting to int throws error, return a bad request status code
+    ctx.res.status = 400
+    return
+
+  plt.hist(x=data)
+
+  # The title, ylabel and xlabel are all optional so check they exist before using them.
+  plt.title(title[0] if title is not None else "")
+  plt.xlabel(xlabel[0] if xlabel is not None else "")
+  plt.ylabel(ylabel[0] if ylabel is not None else "")
+```
+
+We will also want to make the bins for our histogram equal to the range of the histogram. For this we will use numpy, as it will support using floats as well as integers.
+
+```bash
+pipenv install numpy
+```
+
+```python
+import numpy as np
+
+...
+# Get the number of bins as the x-range of the data
+plt.hist(x=data, bins=np.arange(min(data), max(data) + 2, 1))
+...
+```
+
+At this point the plot will be created, however, nothing but a 200 status will be returned to the user. To actually return the data as an image to the user, we will need to first get the image data.
+
+```python
+plt.savefig("plot.png")
```

Comment:
Maybe a buffer could be used to store the image instead of writing it to disk in the container. Something like this? https://github.com/matplotlib/matplotlib/issues/23319